BUILDER_DEV.md

```diff
@@ -1,390 +1,294 @@
+This document captures the steps to start and run a Builder environment for development. The builder environment includes the builder services, as well as the depot web site.
 
-## Bootstrap the OS with required packages
-You need to make sure you have the required packages installed.
-Run the appropriate shell script under the `support/linux` folder to install the packages.
-Refer to [BUILDING.md](./BUILDING.md) doc for the detailed steps.
+There are several ways of creating a Builder dev environment - but supporting all operating systems and environments has proven to be untenable. This document includes one officially supported way of creating a Builder dev environment, and links to unsupported ways of creating the dev environment that you may use at your own risk.
 
-## Create Builder encryption key pair
-If you haven't already done this earlier, you'll need to create a key pair for builder to encrypt credentials.
-Do this via the `hab user key generate bldr` command. This will generate a key pair with the prefix `bldr`
-in your local `hab/cache/keys` directory. Copy the key pair over to some location that is accessible by the running services.
-You will need to specify that location in the config files below.
+Eventually, we will want to harness the power of the Habitat studio to automate a builder dev environment, but this serves as a stop gap for now.
 
-## Create configuration files
-Some capabilities (such as allowing builder permissions, and turning on auto-building when new packages are uploaded to the depot) require passing in a custom config to the Builder services.
+## Officially Supported Dev Environment
 
-Create the following files somewhere on your local filesystem:
+This Dev Environment involves spinning up a virtual machine (most of the core contributors use VMWare, but Virtual Box should work as well) running Ubuntu 17.10 desktop.
 
-`config_api.toml`
-```toml
-[depot]
-path = "/hab/svc/builder-api/data"
-key_dir = "/path/to/bldr-key-pair"
+### Pre-Reqs
+
+* Ubuntu 17.10 Desktop Virtual Machine
+* Access to the "core" origin in [Production Builder](https://bldr.habitat.sh). Ask a core maintainer (or in the [Habitat Slack](http://slack.habitat.sh/)) if you do not have access.
+* Access to the Habitat 1Password account (NOTE - we are working on a solution to make this not required, but it is necessary for the time being)
+* All of the following steps should be run within your Virtual Machine
+* A [Github Personal Access token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) with all repo and all user permissions.
+* The [Habitat Builder App](https://github.com/apps/habitat-builder) installed on your Github account.
+* Open your Sudoers file at /etc/sudoers and remove the "secure_path" section (this is required for the make scripts to run correctly)
+
+```
+$ sudo visudo -f /etc/sudoers
```

Comment:
For those of us who aren't `vi` users, it might be nice to supply an alternative. For me (since emacs isn't already installed) this works well:
```
$ EDITOR=$(which nano) sudoedit /etc/sudoers
```
Or it can be totally automated:
```
$ sudo sed -i.bak '/secure_path/s/^/#/' /etc/sudoers
```